packages/braid-design-system/lib/components/private/buildDataAttributes.ts

```diff
@@ -1,6 +1,68 @@
+import dedent from 'dedent';
+
 export type DataAttributeMap = Record<string, string | number>;
 
-export default (data: DataAttributeMap = {}): DataAttributeMap => {
+interface DataAttributeParams {
+  data?: DataAttributeMap;
+  validateRestProps: Record<string, any> | false;
+}
+
+export default ({ data, validateRestProps }: DataAttributeParams) => {
+  if (process.env.NODE_ENV !== 'production') {
+    if (validateRestProps !== false) {
+      const dataAttrs = Object.keys(validateRestProps).filter((prop) =>
+        prop.startsWith('data-'),
+      );
+
+      const formatter = (v: boolean | string | number) => {
+        if (typeof v === 'boolean') {
+          return v ? '' : `={false}`;
+        }
+
+        if (typeof v === 'number') {
+          return `={${v}}`;
+        }
+
+        return `="${v}"`;
+      };
```

Comment:
Could be moved to the top-level scope since it's a pure function, but it's only run in dev so I'm not fussed. The dev-time perf tho...

Reply:
Yep I typically would have, but left this inside so it gets stripped out of the production bundle along with the entire if block.

Reply:
Oh of course, that makes sense.